proto/cusf/mainchain/v1/wallet.proto

```diff
@@ -0,0 +1,72 @@
+/* CUSF mainchain wallet service */
+
+syntax = "proto3";
+package cusf.mainchain.v1;
+
+import "google/protobuf/wrappers.proto";
+
+import "cusf/mainchain/v1/common.proto";
+
+enum AddressType {
+  ADDRESS_TYPE_UNSPECIFIED = 0;
+  ADDRESS_TYPE_DEFAULT = 1;
+  ADDRESS_TYPE_BECH32 = 2;
+  ADDRESS_TYPE_LEGACY = 3;
+  ADDRESS_TYPE_P2SH_SEGWIT = 4;
+}
+
+service WalletService {
+  rpc BroadcastWithdrawalBundle(BroadcastWithdrawalBundleRequest)
+    returns (BroadcastWithdrawalBundleResponse);
+  rpc CreateBmmCriticalDataTransaction(CreateBmmCriticalDataTransactionRequest)
+    returns (CreateBmmCriticalDataTransactionResponse);
+  rpc CreateDepositTransaction(CreateDepositTransactionRequest)
+    returns (CreateDepositTransactionResponse);
+  rpc CreateNewAddress(CreateNewAddressRequest)
+    returns (CreateNewAddressResponse);
+  // Regtest only
+  rpc GenerateBlocks(GenerateBlocksRequest)
+    returns (GenerateBlocksResponse);
```

Comment:
I think this can be axed. Generating blocks currently happens through the `contrib/signet/miner` script provided by Bitcoin Core, with minor tweaks by us. I'm currently working on a tweak here where we pass in CLI args for extra OP_RETURN messages that we want to add, which will let us do sidechain management. If we instead do generation through the wallet service, I think it'll end up being very hairy and complex

Reply:
This is only for Regtest mode, so the signet script is irrelevant